govcd/landing_zone.go

```diff
@@ -0,0 +1,207 @@
+	}
+
+	// 6. Reload RDE
+	err = createdRdeEntity.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing RDE after resolving: %s", err)
+	}
+
+	result, err := convertRdeEntityToAny[types.SolutionLandingZoneType](createdRdeEntity.DefinedEntity.Entity)
+	if err != nil {
+		return nil, err
+	}
+
+	returnType := SolutionLandingZone{
+		SolutionLandingZoneType: result,
+		vcdClient:               vcdClient,
+		DefinedEntity:           createdRdeEntity,
+	}
+
+	return &returnType, nil
+}
+
+// GetAllSolutionLandingZones retrieves all Solution Landing Zones
+//
+// Note: Up to VCD 10.5.1.1 there can be only a single RDE entry (one SLZ per VCD)
+func (vcdClient *VCDClient) GetAllSolutionLandingZones(queryParameters url.Values) ([]*SolutionLandingZone, error) {
+	allSlzs, err := vcdClient.GetAllRdes("vmware", "solutions_organization", "1.0.0", queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving all SLZs: %s", err)
+	}
+
+	results := make([]*SolutionLandingZone, len(allSlzs))
+	for slzRdeIndex, slzRde := range allSlzs {
+
+		slz, err := convertRdeEntityToAny[types.SolutionLandingZoneType](slzRde.DefinedEntity.Entity)
+		if err != nil {
+			return nil, fmt.Errorf("error converting RDE to SLZ: %s", err)
+		}
+
+		results[slzRdeIndex] = &SolutionLandingZone{
+			vcdClient:               vcdClient,
+			DefinedEntity:           slzRde,
+			SolutionLandingZoneType: slz,
+		}
+	}
+
+	return results, nil
+}
+
+// GetExactlyOneSolutionLandingZone will get single Solution Landing Zone RDE or fail.
+// There can be only one Solution Landing Zone in VCD, but because it is backed by RDE - it can
+// occur that due to some error there is more than one RDE Entity
+func (vcdClient *VCDClient) GetExactlyOneSolutionLandingZone() (*SolutionLandingZone, error) {
+	allSlzs, err := vcdClient.GetAllSolutionLandingZones(nil)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving all Solution Landing Zones: %s", err)
+	}
+
+	return oneOrError("rde", "vmware:solutions_organization:1.0.0", allSlzs)
+}
+
+// GetSolutionLandingZoneById retrieves Solution Landing Zone by ID
+//
+// Note: defined entity ID must be used that can be accessed either by `SolutionLandingZone.Id()`
+// method or directly in `SolutionLandingZone.DefinedEntity.DefinedEntity.ID` field
+func (vcdClient *VCDClient) GetSolutionLandingZoneById(id string) (*SolutionLandingZone, error) {
+	if id == "" {
+		return nil, fmt.Errorf("id must be specified")
+	}
+	rde, err := getRdeById(&vcdClient.Client, id)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDE by ID: %s", err)
+	}
+
+	result, err := convertRdeEntityToAny[types.SolutionLandingZoneType](rde.DefinedEntity.Entity)
+	if err != nil {
+		return nil, err
+	}
+
+	packages := &SolutionLandingZone{
+		SolutionLandingZoneType: result,
+		vcdClient:               vcdClient,
+		DefinedEntity:           rde,
+	}
+
+	return packages, nil
+}
+
+func (slz *SolutionLandingZone) Refresh() error {
```

Comment:
Not sure if it will give much value, but since this is public, maybe we can add docs?

Reply:
Fixed.